CHANGELOG

```diff
@@ -1,3 +1,336 @@
+back into an `odp_event_t`.
+
+* `odp_crypto_result()` extracts the `odp_crypto_packet_result_t` struct that
+contains the crypto metadata associated with an `odp_packet_t` of event
+subtype `ODP_EVENT_PACKET_CRYPTO`. This struct provides a summary bit that
+says whether the operation completed successfully as well as
+`odp_crypto_op_status_t` fields for the `cipher_status` and `auth_status` if a
+problem was detected.
+
+===== Classification Random Early Detection (RED) Support
+Random Early Detection (RED) provides a means for input HW to ensure that
+traffic is treated fairly under conditions of temporary resource overload due
+to excessive inbound traffic. ODP RED support provides the ability to measure
+and respond to resource pressure on either pools or queues, and only provides
+flexibility in how such conditions are to be processed. They can result in
+input packet drops or backpressure being indicated by transmitting pause
+frames, depending on the underlying platform capabilities.
+
+The `odp_cls_capability_t` struct returned by the `odp_cls_capability()` API
+has been expanded to cover this support.
+
+===== Time difference in nanoseconds
+The new `odp_time_diff_ns()` API permits the delta between two `odp_time_t`
+values to be computed in a single call.
+
+===== PktIO Maximum Frame Lengths
+The `odp_pktio_mtu()` API is deprecated and replaced by two new APIs:
+`odp_pktin_maxlen()` and `odp_pktout_maxlen()`. These return the maximum
+sized Ethernet packets that are supported for RX and TX processing,
+respectively, on a given `odp_pktio_t`.
+
+===== Pool subparameter support
+The `odp_pool_param_t` structure has been expanded to provide more flexibility
+to support platforms that are able to offer multiple segment sizes within a
+single pool. This can lead to greater storage efficiency. These are called
+subparameters and implementations supporting up to 7 of these are accommodated
+with these extensions.
+
+The `odp_pool_capability_t` structure is expanded to return the number of
+subparameters supported by this implementation. The application, in turn,
+specifies its expected packet size and number distribution in the
+`odp_pool_pkt_subparam_t` structure that is part of the `odp_pool_param_t`
+used to define the characteristics of `ODP_POOL_PACKET` pools.
+
+This is fully compatible with previous packet pool support since ODP
+implementations are under no obligation to support pool subparameters and
+these, when present, are advisory in nature. They simply serve to allow the
+application to better communicate its expected packet distribution within a
+pool so that the ODP implementation may better optimize storage use in the
+pool.
+
+===== Checksum support
+Checksum processing support has been formalized with the addition of APIs for
+determining packet checksum status, controlling packet checksum processing,
+and computing checksum partial sums for both packets and general buffers.
```

Comment:
This is not quite true. There is no support for computing partial checksum for a packet. Current API returns information about checksum precomputed by hardware when receiving a packet.
Computing partial checksum over a buffer is supported.

Reply:
`odp_packet_ones_comp()` is specifically for computing partial checksums, no?

Reply:
```
 * Returns 16-bit ones' complement sum that was calculated over a portion of
 * packet data during a packet processing operation (e.g. packet input or
 * IPSEC offload). The data range is output with 'range' parameter, and usually
 * includes IP payload (L4 headers and payload). When 'range.length' is zero,
 * the sum has not been calculated. In case of odd number of bytes,
 * calculation uses a zero byte as padding at the end. The sum may be used as
 * part of e.g. UDP/TCP checksum checking, especially with IP fragments.
```

Reply:
OK, I missed that. Thanks. Will fix in v3.